blueprints/tr-load-tests.md

```diff
@@ -0,0 +1,296 @@
+What are the column data types and modifiers?
+What are the FK references and constraints?
+-->
+No Data Model impact is anticipated.
+
+### Cache Config Impact
+<!--
+*How* will this impact ORT?
+-->
+No Cache Config impact is anticipated.
+
+### Traffic Monitor Impact
+<!--
+*How* will this impact Traffic Monitor?
+
+Will new profile parameters be required?
+-->
+No Traffic Monitor impact is anticipated.
+
+### Traffic Router Impact
+<!--
+*How* will this impact Traffic Router?
+
+Will new profile parameters be required?
+How will the CRConfig be changed?
+How will changes in Traffic Ops data be reflected in the CRConfig?
+Will Traffic Router remain backwards-compatible with old CRConfigs?
+Will old Traffic Routers remain forwards-compatible with new CRConfigs?
+-->
+The addition of the TR Ultimate Test Harness themselves will not change Traffic Router functionality in any way. For visibility, however, the TR Ultimate Test Harness should reside in a directory within the `traffic_router` directory. This will be the first time since 545929f7cc that Golang sources will exist in the `traffic_router` directory, so any assumption that all sources within the `traffic_router` directory directly impact Traffic Router's ability to compile should be abandoned.
+
+The TR Ultimate Test Harness should not be included in the Traffic Router RPM, as it is meant to be run on a host separate from Traffic Routers.
+
+### Traffic Stats Impact
+<!--
+*How* will this impact Traffic Stats?
+-->
+No Traffic Stats impact is anticipated.
+
+### Traffic Vault Impact
+<!--
+*How* will this impact Traffic Vault?
+
+Will there be any new data stored in or removed from Riak?
+Will there be any changes to the Riak requests and responses?
+-->
+No Traffic Vault impact is anticipated.
+
+### Documentation Impact
+<!--
+*How* will this impact the documentation?
+
+What new documentation will be required?
+What existing documentation will need to be updated?
+-->
+Instructions for using the Traffic Router Ultimate Test Harness should be added to the documentation. This should include:
+- Small rationale for inclusion of TR Ultimate Test Harness
+- Setup instructions
+    * The permissions that a user running the Traffic Router Test Harness should have:
+        - CDN snapshots
+        - CDN information
+        - information about Traffic Router-type Servers in those CDNs
+        - Type information
+        - Delivery Service information
+    - Documentation of each CLI option
+    - Example commands
+
+### Testing Impact
+<!--
+*How* will this impact testing?
+
+What is the high-level test plan?
+How should this be tested?
+Can this be tested within the existing test frameworks?
+How should the existing frameworks be enhanced in order to test this properly?
+-->
+#### Load Tests
+The Router Ultimate Test Harness should include a load test for HTTP-routed Delivery Services and for DNS-routed Delivery Services
+
+##### Load Test Options
```

Comment:
Are these going to be CLI options? Will there be a config file? Some of these options seem more generic (e.g. CDN name), while some of them are load-test specific (e.g. req/s threshold). If we extend the test harness with more than just load testing, would each different type of test have its own config options and/or file?

Reply:
> Are these going to be CLI options?

No

> Will there be a config file?

Config file structure clarified in c0bd076c46

> If we extend the test harness with more than just load testing, would each different type of test have its own config options and/or file?

Each test can specify its own options, but options applying to more than one test can go under `"all"`.